### PR TITLE
Added popularity sort 

### PR DESCRIPTION
**Pull request by Balkmeister**
You can now sort by popularity (cube inclusion percentage)

### CODE REVIEW

**Comment by lunakv** [comment]
Typographic note: the ranges should all be denoted by en dashes, not hyphens.

File: models/cardHistory.js [hunk 1]
```diff
@@ -61,8 +61,8 @@ const cardHistorySchema = mongoose.Schema({
 cardHistorySchema.index({ oracleId: 1, 'current.rating': 1 });
 cardHistorySchema.index({ oracleId: 1, 'current.elo': -1 });
 cardHistorySchema.index({ oracleId: 1, 'current.picks': -1 });
-cardHistorySchema.index({ oracleId: 1, 'current.cubes': -1 });
-
+cardHistorySchema.index({ oracleId: 1, 'current.total': 1 });
+cardHistorySchema.index({ oracleId: 1 });
```
Comment (ruler501) [inline models/cardHistory.js:65]: This is redundant, you can index on prefixes of other indices.

File: serverjs/updatecards.js [hunk 2]
```diff
@@ -642,6 +643,7 @@ function convertCard(card, isExtra) {
     tix: card.prices.tix ? parseFloat(card.prices.tix) : null,
   };
   newcard.elo = catalog.elodict[name] || 1200;
+  newcard.popularity = Math.round(catalog.poplularitydict[card.oracle_id] * 1000.0) / 10 || 0;
```
Comment (ruler501) [inline serverjs/updatecards.js:646]: Why not store this as full precision?

File: src/utils/Sort.js [hunk 3]
```diff
@@ -640,6 +643,17 @@ export function cardGetLabels(card, sort, showOther) {
...
+    const popularity = Math.round(card.details.popularity);
+    if (popularity < 1) ret = ['0-1%'];
+    else if (popularity < 2) ret = ['1-2%'];
+    else if (popularity < 5) ret = ['3-5%'];
+    else if (popularity < 8) ret = ['5-8%'];
+    else if (popularity < 12) ret = ['8-12%'];
+    else if (popularity < 20) ret = ['12-20%'];
+    else if (popularity < 30) ret = ['20-30%'];
+    else if (popularity < 50) ret = ['30-50%'];
+    else if (popularity < 100) ret = ['50-100%'];
```
Comment (ruler501) [inline src/utils/Sort.js:656]: Just for completeness sake (and local testing) this should be `<=` for the highest bracket.

**Review by ruler501 — CHANGES_REQUESTED** [review]

**Pull request merged**